solarforecastarbiter/metrics/calculator.py

```diff
@@ -249,11 +249,45 @@ def calculate_deterministic_metrics(processed_fx_obs, categories, metrics,
                 metric_vals.append(datamodel.MetricValue(
                     category, metric_, str(cat), res))
 
-    out['values'] = tuple(metric_vals)
+    out['values'] = _sort_metrics_vals(
+        metric_vals, datamodel.ALLOWED_DETERMINISTIC_METRICS)
     calc_metrics = datamodel.MetricResult.from_dict(out)
     return calc_metrics
 
 
+def _sort_metrics_vals(metrics_vals, mapping):
+    """
+    Parameters
+    ----------
+    metrics_vals : list
+        Elements are datamodel.MetricValue
+    mapping : dict
+        Metrics mapping defined in datamodel.
+
+    Returns
+    -------
+    tuple
+        Sorted elements. Sorting order is:
+            * Category
+            * Metric
+            * Index
+            * Value
+    """
+    metric_ordering = list(mapping.keys())
+    category_ordering = list(datamodel.ALLOWED_CATEGORIES.keys())
```

Comment:
I feel like we should comment that this ordering is only necessarily preserved in python >= 3.6 with dicts ordered by default now

Reply:
even though we're 3.7+ and use dataclasses?

Reply:
ah forgot about the dataclasses that will break 3.6 anyway, so not necessary